internal/config/config.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"semangit/internal/plugins"
+	"semangit/internal/plugins/base"
+	"semangit/internal/plugins/versionanalyzer"
+	"strings"
+)
+
+const RevisionNone = ""
+
+type Config struct {
+	RepoDir                   string                     `json:"repo_dir,omitempty"`
+	SrcRevision               string                     `json:"src_revision,omitempty"`
+	DestRevision              string                     `json:"dest_revision,omitempty"`
+	ActiveVersionAnalyzerName string                     `json:"active_version_analyzer_name,omitempty""`
```

Comment:
As we have used **Current** in `GetCurrentVersionAnalyzerArgumentValues`, for the sake of consistency I recommend `CurrentVersionAnalyzerName` as well.